### ISSUE TITLE
Parsing fails if option value clashes with sub-command name

### ISSUE THREAD

**Issue by jnehlmeier** [remkop/picocli#2065]
If you have something like 
```
./cli import --database-name=persons persons --csv=/tmp/persons.csv
```
the cli fails with
```
Expected parameter for option '--database-name' but found 'persons'
```

I believe the option value `persons` is already treated as sub-command name in the parsing code and thus the CLI complains about a missing option parameter (the option is required).

**Comment by remkop** [comment]
Yes that’s the default parser behavior but that can be changed with configuration:

https://picocli.info/#_enable_consuming_option_names_or_subcommands

**Comment by jnehlmeier** [comment]
Oh didn't know that. Thanks for pointing it out. Is there a reason why it is parsed this way by default? As far as I know a `string` field has an arity of 1 by default so it should be pretty clear to parse the above command line correctly.

**Comment by remkop** [comment]
The reason is that picocli doesn’t distinguish between 

```
--option=value 
```

and

```
--option value 
```

So given `--option subcommand`, if the string `subcommand`is a valid subcommand, picocli will by default not interpret it as an option parameter but as a subcommand, and will warn the user that they omitted the parameter for `--option`. 

I believe this is the most common use case so I decided to make this the default.

**Comment by jnehlmeier** [comment]
I see. Thanks for the explanation and work you put into the library.
